Source/IntegrationTests/TestFiles/LitTests/LitTest/reveal/revealInBlock.dfy

```diff
@@ -0,0 +1,71 @@
+// echo ''
+// NORUN: ! %verify --type-system-refresh --allow-axioms --bprint:%t.bpl --isolate-assertions --boogie "/printPruned:%S/pruned" %s > %t
+// NORUN: %diff "%s.expect" "%t"
+
+function P(x: int): bool {
+  true
+}
+
+function Q(): int 
+  requires P(0) {
+  3
+}
+
+method IfScope() {
+  hide *;
+  if (*) {
+    reveal P;
+  } else {
+    reveal P;
+  }
+  assert P(2); // error, since the previous two reveal statements are out of scope
+}
+
+method MatchStatementScope(x: int) {
+  hide *;
+  match x {
+    case 0 =>
+      reveal P;
+      assert P(0);
+    case _ =>
+      assert P(0); // error
+  }
+  assert P(1); // error
+}
+
+method BlockScope() {
+  hide *;
+  {
+    reveal P;
+    assert P(0);
+  }
+  assert P(1); // error
```

Comment:
Could you add `assert P(0);` after the block so that we capture the current verification behavior, whether or not it escapes?